apollo-federation/src/query_graph/build_query_graph.rs

```diff
@@ -2333,42 +2344,66 @@ fn resolvable_key_applications<'doc>(
 }
 
 fn parse_context(field: &str) -> Result<(String, String), FederationError> {
```

Comment:
Bugs:
- The negative look-aheads in the JS regex were dropped here when we should be accounting for them (my guess is it's because the `regex` crate doesn't natively support them). For this particular case, the negative look-aheads are effectively used to prevent backtracking. We can do that manually without negative look-aheads if we split the regex into smaller regexes.

Reply:
No worries! A test makes sense, added it in [d71dff0](https://github.com/apollographql/router/pull/6380/commits/d71dff0f9845031253f11fd2e5aaf6d4d0573279).